src/luarocks/fs/win32/tools.lua

```diff
@@ -238,16 +244,19 @@ function tools.set_permissions(filename, mode, scope)
       if not ok then
          return false, "Could not take ownership of the given file"
       end
+      local username = os.getenv('USERNAME')
       -- Grant the current user the proper rights
-      ok = fs.execute_quiet(vars.ICACLS .. " " .. fs.Q(filename) .. " /inheritance:d /grant:r \"%USERNAME%\":" .. perms)
+      ok = fs.execute_quiet(vars.ICACLS .. " " .. fs.Q(filename) .. " /inheritance:d /grant:r " .. fs.Q(username) .. ":" .. perms)
```

Comment:
Why do you need this change?

Reply:
This more for explicitly use the same value in this command and in the if condition later.